src/Evolution/SlopeLimiters/Minmod.hpp

```diff
@@ -0,0 +1,440 @@
+          const auto dir = Direction<VolumeDim>(d, side);
+          const bool has_neighbors = (externals.find(dir) == externals.end());
+          if (has_neighbors) {
+            // Compute an effective element-center-to-neighbor-center distance
+            // that accounts for the possibility of different refinement levels
+            // or discontinuous maps (e.g., at Block boundaries). By using this
+            // effective distance, a linear function at a refinement or Block
+            // boundary will still appear smooth to the limiter (which works in
+            // the logical coordinates). This is not an attempt to construct a
+            // slope, but rather an attempt to adjust to difference-of-means to
+            // avoid limiter actuations on smooth solutions.
+            // Note that this is not "by the book" Minmod, but is an attempt to
+            // generalize it to work on non-uniform grids.
+            const double half_effective_distance =
+                0.5 *
+                (1.0 + neighbor_sizes.at(dir).get(d) / element_size.get(d));
+            return (side == Side::Lower ? -1.0 : 1.0) *
+                   (neighbor_tensors.at(dir)[i] - u_mean) /
+                   half_effective_distance;
+          } else {
+            return 0.0;
+          }
+        };
+        const double diff_upper = difference_of_means_to_neighbor(Side::Upper);
+        const double diff_lower = difference_of_means_to_neighbor(Side::Lower);
+
+        // Results from SpECTRE paper (https://arxiv.org/abs/1609.00098) used
+        // minmod_tvbm(..., 0.0), rather than minmod_tvbm(..., tvbm_scale)
+        const double v_lower =
+            u_mean -
+            minmod_tvbm(u_mean - u_lower, diff_lower, diff_upper, tvbm_scale)
+                .value;
+        const double v_upper =
+            u_mean +
+            minmod_tvbm(u_upper - u_mean, diff_lower, diff_upper, tvbm_scale)
+                .value;
+        // Value of epsilon from Hesthaven & Warburton, Chapter 5, in the
+        // SlopeLimitN.m code sample.
+        const double eps = 1.e-8;
+        if (fabs(v_lower - u_lower) > eps or fabs(v_upper - u_upper) > eps) {
+          u_needs_limiting = true;
+          break;
+        }
+      }
+
+      if (not u_needs_limiting) {
+        // Skip the limiting step for this tensor component
+        continue;
+      }
+    }
+
+    const DataVector u_lin = linearize(u, mesh);
+    bool this_component_was_limited = false;
+    auto u_limited_slopes = make_array<VolumeDim>(0.0);
+
+    for (size_t d = 0; d < VolumeDim; ++d) {
+      const double u_lower =
+          mean_value_on_boundary(u_lin, mesh, d, Side::Lower);
+      const double u_upper =
+          mean_value_on_boundary(u_lin, mesh, d, Side::Upper);
+      // In logical coordinates, the width of the element is 2.0
+      const double local_slope = (u_upper - u_lower) / 2.0;
+
+      const auto approx_slope_to_neighbor = [
```

Comment:
Unless I've missed something this lambda is the same as the one above except for a factor of 2.  Can it be pulled out so it is not repeated twice?